docs/guides/drivers/kubernetes.md

```diff
@@ -107,28 +83,54 @@ kube0-6977cdcb75-rkc6b   1/1     Running   0          8s
 kube0-6977cdcb75-vb4ks   1/1     Running   0          8s
 kube0-6977cdcb75-z4fzs   1/1     Running   0          8s
 ```
-    
+
 Additionally, you can use the `loadbalance=(sticky|random)` option to control
-the load-balancing behavior when there are multiple replicas. While `random`
-should selects random nodes from the available pool, which should provide
-better balancing across all replicas, `sticky` (the default) attempts to
-connect the same build performed multiple times to the same node each time,
-ensuring better local cache utilization.
+the load-balancing behavior when there are multiple replicas. `random` selects
+random nodes from the node pool, providing an even workload distribution across
+replicas. `sticky` (the default) attempts to connect the same build performed
+multiple times to the same node each time, ensuring better use of local cache.
+
+For more information on scalability, see the options for
+[buildx create](https://docs.docker.com/engine/reference/commandline/buildx_create/#driver-opt).
+
+## Node assignment
 
-For more information on scalability, see the options for [buildx create](https://docs.docker.com/engine/reference/commandline/buildx_create/#driver-opt).
+The Kubernetes driver allows you to control the scheduling of BuildKit pods
+using the `nodeSelector` and `tolerations` driver options.
+
+The value of the `nodeSelector` parameter is a comma-separated string of
+key-value pairs, where the key is the node label and the value is the label
+text. For example: `"nodeselector=kubernetes.io/arch=arm64"`
+
+The `tolerations` parameter is a semicolon-separated list of taints. It accepts
+the same values as the Kubernetes manifest. Each `tolerations` entry specifies a
+taint key and the value, operator, or effect. For example:
+`"tolerations=key=foo,value=bar;key=foo2,operator=exists;key=foo3,effect=NoSchedule"`
+
+The syntax for these parameters is slightly different compared to other driver
+options. You must wrap both `nodeSelector` and `tolerations` in double quotes.
+For example:
+
+```console
+$ docker buildx create \
+  --bootstrap \
+  --name=kube \
+  --driver=kubernetes \
+  --driver-opt="nodeselector=label=value","tolerations=key=key1,value=value1"
+```
```

Comment:
Hm this isn't right.

The final line should be `--driver-opt='"nodeselector=label=value","tolerations=key=key1,value=value1"'` (single quotes around the option value), or even `'--driver-opt="nodeselector=label=value","tolerations=key=key1,value=value1"'` (single quotes around the whole option).

Otherwise, the double quotes will actually get removed by `/bin/sh`, so buildkit sees the same values as before. We need `/bin/sh` to remove the single quotes, so buildkit actually gets the double quoted values correctly (yes quoting rules are horrible :tada:).

Phrasing this is hard as well - these options don't have explicitly different syntax from the others, it's that because they may contain commas, these would otherwise get parsed incorrectly. The CSV parser lets us explicitly group a field that may contain commas by using double quotes around the entire field.

I wonder if we might want to rework the code for some of the details of this syntax to cover the edge cases here, and at least be a little more permissive with how we do the parsing (making sure we don't silently drop ignored csv options would also be a good step to help error correct here).